lib/timeline_records.rb

```diff
@@ -0,0 +1,298 @@
+# frozen_string_literal: true
+
+class TimelineRecords
+  DEFAULT_DETAILS_CONFIG = {
+    cohort_imports: [],
+    class_imports: [],
+    patient_sessions: %i[session_id],
+    school_moves: %i[school_id source],
+    school_move_log_entries: %i[school_id user_id],
+    consents: %i[response route],
+    triages: %i[status performed_by_user_id],
+    vaccination_records: %i[outcome session_id]
+  }.freeze
+
+  AVAILABLE_DETAILS_CONFIG = {
+    cohort_imports: %i[rows_count status uploaded_by_user_id],
+    class_imports: %i[rows_count year_groups uploaded_by_user_id session_id],
+    patient_sessions: %i[session_id],
+    school_moves: %i[source school_id home_educated],
+    school_move_log_entries: %i[user_id school_id home_educated],
+    consents: %i[
+      programme_id
+      response
+      route
+      parent_id
+      withdrawn_at
+      invalidated_at
+    ],
+    triages: %i[status performed_by_user_id programme_id invalidated_at],
+    vaccination_records: %i[
+      outcome
+      performed_by_user_id
+      programme_id
+      session_id
+      vaccine_id
+    ]
+  }.freeze
+
+  DEFAULT_AUDITS_CONFIG = {
+    include_associated_audits: true,
+    include_filtered_audit_changes: false
+  }.freeze
+
+  ALLOWED_AUDITED_CHANGES = %i[
+    patient_id
+    session_id
+    programme_id
+    vaccine_id
+    organisation_id
+    school_id
+    gp_practice_id
+    uploaded_by_user_id
+    performed_by_user_id
+    user_id
+    parent_id
+    status
+    outcome
+    response
+    route
+    date_of_death_recorded_at
+    restricted_at
+    invalidated_at
+    withdrawn_at
+    rows_count
+    year_groups
+    home_educated
+    source
+  ].freeze
+
+  def initialize(patient_id, detail_config: {}, audit_config: {})
+    @patient = Patient.find(patient_id)
+    @patient_id = patient_id
```

Comment:
```suggestion
  def initialize(patient, detail_config: {}, audit_config: {})
    @patient = patient
    @patient_id = patient.id
```

It looks like everywhere we call `TimelineRecords.new` we've got a `Patient` available so we don't need to fetch it each time.